awx/ui_next/src/screens/Setting/LDAP/LDAPEdit/LDAPEdit.jsx

```diff
@@ -1,25 +1,218 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
-import { withI18n } from '@lingui/react';
-import { t } from '@lingui/macro';
-import { Button } from '@patternfly/react-core';
-import { CardBody, CardActionsRow } from '../../../../components/Card';
-
-function LDAPEdit({ i18n }) {
+import React, { useCallback, useEffect } from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { Formik } from 'formik';
+import { Form } from '@patternfly/react-core';
+import { CardBody } from '../../../../components/Card';
+import ContentError from '../../../../components/ContentError';
+import ContentLoading from '../../../../components/ContentLoading';
+import { FormSubmitError } from '../../../../components/FormField';
+import { FormColumnLayout } from '../../../../components/FormLayout';
+import { useSettings } from '../../../../contexts/Settings';
+import { RevertAllAlert, RevertFormActionGroup } from '../../shared';
+import {
+  BooleanField,
+  ChoiceField,
+  EncryptedField,
+  InputField,
+  ObjectField,
+} from '../../shared/SharedFields';
+import { formatJson } from '../../shared/settingUtils';
+import useModal from '../../../../util/useModal';
+import useRequest from '../../../../util/useRequest';
+import { SettingsAPI } from '../../../../api';
+
+function LDAPEdit() {
+  const history = useHistory();
+  const { isModalOpen, toggleModal, closeModal } = useModal();
+  const { PUT: options } = useSettings();
+  const {
+    params: { category },
+  } = useRouteMatch('/settings/ldap/:category/edit');
+  const ldapCategory =
+    category === 'default' ? 'AUTH_LDAP_' : `AUTH_LDAP_${category}_`;
```

Comment:
This is cool. The mutli-ldap subcategories were tough to deal with in the classic ui. Generating some route keys for the different ldap configs and using `ldapCategory` to reference the fields solves that problem. :+1: